packages/react/src/components/Tree/Tree.tsx

```diff
@@ -52,6 +53,13 @@ export interface TreeProps extends UIComponentProps, ChildrenComponentProps {
    * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
    */
   renderItemTitle?: ShorthandRenderFunction
+
+  /** Called when activeIndex changes.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onActiveIndexChange?: ComponentEventHandler<TreeProps & { activeIndex: number[] | number }>
```

Comment:
```suggestion
  onActiveIndexChange?: ComponentEventHandler<TreeProps>
```
The `TreeProps` interface already contains the `activeIndex`